### ISSUE TITLE
Changes to imported less files doesn't cause cache miss

### ISSUE THREAD

**Issue by ropez** [jsdf/browserify-incremental#12]
When used with [lessify](https://www.npmjs.com/package/node-lessify)

Changes to less files included in an `@import` statement doesn't cause the output to be updated. You have to touch the main less file to trigger an update.

Example:
```
# main.less
@import "foo.less";

# foo.less
body { background: blue }

# main.coffee
require('main.less')
```

When a cache exists, chaning `foo.less` doesn't cause the resulting CSS in `main.coffee` to change. I have to run `touch main.less` to cause an update.

**Comment by jsdf** [comment]
Thanks for reminding me of this

relates to https://github.com/jsdf/browserify-incremental/issues/2

**Comment by jsdf** [comment]
Okay firstly for this to work, lessify would need to emit a 'file' event for each less file which is required from a particular module (js/coffee/etc file) and all of its less imports. The brfs transform has [an example of this](https://github.com/substack/brfs/blob/master/index.js#L96).

**Comment by ropez** [comment]
No problem.

Is there anything I can do to help. Check if this can be done in lessify, and do a PR if necessary?

**Comment by jsdf** [comment]
Yep, that would definitely help.

**Comment by ropez** [comment]
I figured it should be as simple as this: https://github.com/snaptv/lessify/commit/f6ee2d68811a4a428859d56ce78012536fe4ded6

Do we actually need to emit an event for each imported file, or can we allow the argument to contain the whole array in a single event? I'm not very familiar with the API, I'm not sure if there's an "official" spec for this 'file' event.

I haven't sent a PR yet, waiting for feedback here first.

(btw, I found out there are two lessify modules, [node-lessify](https://www.npmjs.com/package/node-lessify) and [lessify](https://github.com/dstokes/lessify). I'm using the latter, not the one I linked to in the original post)

**Comment by jsdf** [comment]
The 'spec' is 'whatever watchify does', which you can see [here](https://github.com/substack/watchify/blob/master/index.js#L76-L80). browserify-incremental fulfils the same role as watchify here, and lessify would be doing the equivalent of brfs. Seeing watchify and brfs are written by @substack, that's about as close as there is to a spec for this particular mechanism.

**Comment by ropez** [comment]
Ok, I've tested this patched lessify with watchify, and it does the trick. I'm sending a pull request to lessify.

**Comment by ropez** [comment]
So, the missing pieces have now been merged:

https://github.com/dstokes/lessify/pull/7
https://github.com/jsdf/browserify-cache-api/pull/1

What remains to be able to use this feature is to update the dependency here, and publish new versions to npm. I don't think I can PR tagging and uploading, any chance of a new version in the nearest future?

**Comment by jsdf** [comment]
published 3.0.0

**Comment by ropez** [comment]
Thank you
